examples/nlp/text_classification/conf/config.yaml

```diff
@@ -0,0 +1,81 @@
+# Sentence classification with pretrained BERT models
+
+io:
+  work_dir: ./outputs
+  checkpoint_dir: null
+  save_epoch_freq: 1
+  save_step_freq: -1
+
+pl:
+  trainer:
+    gpus: 1 # the number of gpus, 0 for CPU
+    num_nodes: 1
+    max_epochs: 10
+    max_steps: null # precedence over max_epochs
+    accumulate_grad_batches: 1 # accumulates grads every k batches
+    amp_level: O0 # O1/O2 for mixed precision
+    distributed_backend: ddp
+
+model:
+  data_dir: ??? # /path/to/data
+  class_balancing: null # or weighted_loss
+
+  train_ds:
+    prefix: train
+    batch_size: 32
+    shuffle: true
+    num_samples: -1
+    num_workers: 2
+    pin_memory: false
+    use_cache: true
+    drop_last: true
+
+  validation_ds:
+    prefix: dev
+    batch_size: 32
+    shuffle: false
+    num_samples: -1
+    num_workers: 2
+    pin_memory: false
+    use_cache: true
+    drop_last: true
+
+  language_model:
+    pretrained_model_name: roberta-base
+    max_seq_length: 36
+    bert_checkpoint: null
+    bert_config: null
+    tokenizer: null # only used if using custom checkpoint
+    tokenizer_model: null # only used if tokenizer is sentencepiece
+    do_lower_case: true # true for uncased models, false for cased models
+  
+  head:
+    num_output_layers: 2
+    fc_dropout: 0.1
+
+  optim:
+    name: adam
+    lr: 2e-5
+    args:
+      name: auto
+      params:
+        weight_decay: 0.01
+
+    sched:
```

Comment:
A scheduler requires an optimizer in it's constructor, and a scheduler directly works on that optimizer. It is necessary to bind 1 scheduler to 1 optimizer, as that's how PTL requires it. Also, it makes more sense to have it like this, in case we need more than one optimizer and scheduler (say gans).

Reply:
@titu1994 gave a good explanation below. Also I added a TextClassificationOptimizationConfig to make it more clear in Python.

Reply:
If there is not a way to separate them, then it should be OK.